File: swapi-angular/src/app/api/models/CharacterDTO.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { SearchItemDTO } from './SearchItemDTO';

export type CharacterDTO = {
    birth_year?: string;
    eye_color?: string;
    hair_color?: string;
    skin_color?: string;
    gender?: string;
    height?: string;
    homeworld?: string;
    homeworldItem?: SearchItemDTO;
    mass?: string;
    name?: string;
    edited?: string;
    created?: string;
    url?: string;
    films?: Array<string>;
    filmItems?: Array<SearchItemDTO>;
    species?: Array<string>;
    speciesItems?: Array<SearchItemDTO>;
    starships?: Array<string>;
    starshipItems?: Array<SearchItemDTO>;
    vehicles?: Array<string>;
    vehicleItems?: Array<SearchItemDTO>;
};

File: swapi-server/src/main/java/de/lkaktus/entities/CharacterDTO.java
package de.lkaktus.entities;

import com.fasterxml.jackson.annotation.JsonProperty;

public record CharacterDTO(
        @JsonProperty("birth_year") String birthYear,
        @JsonProperty("eye_color") String eyeColor,
        String gender,
        @JsonProperty("hair_color") String hairColor,
        String height,
        String homeworld,
        SearchItemDTO homeworldItem,
        String mass,
        String name,
        @JsonProperty("skin_color") String skinColor,
        String edited,
        String created,
        String url,
        String [] films,
        SearchItemDTO [] filmItems,
        String [] species,
        SearchItemDTO [] speciesItems,
        String [] starships,
        SearchItemDTO [] starshipItems,
        String [] vehicles,
        SearchItemDTO [] vehicleItems
        ) {
}
